plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/JavaToJKTreeBuilder.kt

```diff
@@ -403,9 +402,16 @@ class JavaToJKTreeBuilder(
                         .qualified(qualifier ?: JKThisExpression(JKLabelEmpty()))
                 }
 
-                symbol is JKMethodSymbol ->
-                    JKCallExpressionImpl(symbol, arguments.toJK(), typeArguments)
-                        .qualified(qualifier)
+                symbol is JKMethodSymbol -> {
+                    val fqName = target?.kotlinFqName.toString()
+                    if ((fqName.endsWith("charAt") || fqName.endsWith("get")) && arguments.expressionCount == 1) {
```

Comment:
To properly convert 'get' method calls to operator index access, we have to know that either:

1. This 'get' is a Java method that will not be converted to Kotlin in this session
2. This is a Kotlin method that has an 'operator' modifier and a proper signature or it overrides such a method (see logic in ReplaceGetOrSetInspection)

And `fqName.endsWith("get")` is not correct: it has to be exactly 'get'.

If these conditions are hard to support, I suggest to concentrate on the array case only (at first).